Build error with git master (2023-03-10)

### Operating System Info

Other

### Other OS

Arch Linux

### OBS Studio Version

Git

### OBS Studio Version (Other)

_No response_

### OBS Studio Log URL

It does not even get to install, so no log available

### OBS Studio Crash Log URL

_No response_

### Expected Behavior

Complete build, then an install

### Current Behavior

When building OBS studio from git, it fails on this part:

```
[ 52%] Building C object plugins/rtmp-services/CMakeFiles/rtmp-services.dir/rtmp-common.c.o
/ad/OS/aur/obs-studio-git/src/obs-studio-git/deps/media-playback/media-playback/decode.c: In function ‘mp_decode_next’:
/ad/OS/aur/obs-studio-git/src/obs-studio-git/deps/media-playback/media-playback/decode.c:452:17: error: ‘pkt_duration’ is deprecated [-Werror=deprecated-declarations]
  452 |                 int64_t duration = d->in_frame->pkt_duration;
      |                 ^~~~~~~
In file included from /usr/include/libavutil/hwcontext.h:23,
                 from /usr/include/libavcodec/codec.h:27,
                 from /usr/include/libavformat/avformat.h:313,
                 from /ad/OS/aur/obs-studio-git/src/obs-studio-git/deps/media-playback/media-playback/decode.h:31,
                 from /ad/OS/aur/obs-studio-git/src/obs-studio-git/deps/media-playback/media-playback/decode.c:17:
/usr/include/libavutil/frame.h:631:13: note: declared here
  631 |     int64_t pkt_duration;
      |             ^~~~~~~~~~~~
cc1: all warnings being treated as errors
make[2]: *** [plugins/obs-ffmpeg/CMakeFiles/obs-ffmpeg.dir/build.make:272: plugins/obs-ffmpeg/CMakeFiles/obs-ffmpeg.dir/__/__/deps/media-playback/media-playback/decode.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 52%] Building C object plugins/rtmp-services/CMakeFiles/rtmp-services.dir/rtmp-custom.c.o
```

### Steps to Reproduce

1. On Arch linux, run yay -S obs-studio-git


### Anything else we should know?

Arch system is fully updated, with rolling release.
System info:
```
Operating System: Arch Linux 
KDE Plasma Version: 5.27.2
KDE Frameworks Version: 5.103.0
Qt Version: 5.15.8
Kernel Version: 6.2.2-AMD-znver2 (64-bit)
Graphics Platform: X11
Processors: 8 × AMD Ryzen 7 4700U with Radeon Graphics
Memory: 36.9 GiB of RAM
Graphics Processor: AMD Radeon Graphics
Manufacturer: LENOVO
Product Name: 20T8001JUS
System Version: ThinkPad E15 Gen 2
```

--- a/deps/media-playback/media-playback/decode.c
+++ b/deps/media-playback/media-playback/decode.c
@@ -446,7 +446,7 @@ bool mp_decode_next(struct mp_decode *d)
 				av_rescale_q(d->in_frame->best_effort_timestamp,
 					     d->stream->time_base,
 					     (AVRational){1, 1000000000});
-#if LIBAVUTIL_VERSION_MAJOR >= 57 && LIBAVUTIL_VERSION_MINOR >= 30
+#if LIBAVUTIL_VERSION_INT >= AV_VERSION_INT(57, 30, 100)
 		int64_t duration = d->in_frame->duration;
 #else
 		int64_t duration = d->in_frame->pkt_duration;

